docs/real-time-analytics/location-based-alerting/process.sql

```diff
@@ -0,0 +1,85 @@
+SET 'auto.offset.reset' = 'earliest';
+
+-- Table of merchant data including the calculated geohash
+CREATE TABLE merchant_locations (
+  id INT PRIMARY KEY,
+  description VARCHAR,
+  latitude DECIMAL(10,7),
+  longitude DECIMAL(10,7),
+  geohash VARCHAR
+)
+WITH (
+    KAFKA_TOPIC='merchant-locations', 
+    VALUE_FORMAT='JSON',
+    PARTITIONS = 6
+);
+
+-- A table to lookup merchants based on the a substring (precision) of the geohash
+CREATE TABLE merchants_by_geohash
+WITH (
+    KAFKA_TOPIC='merchant-geohash', 
+    FORMAT='JSON', 
+    PARTITIONS=6
+) AS 
+SELECT 
+  SUBSTRING(geohash, 1, 6) AS geohash, 
+  COLLECT_LIST(id) as id_list
+FROM merchant_locations
+GROUP BY SUBSTRING(geohash, 1, 6);
+
+-- Stream of user location data including the calculated geohash
+CREATE STREAM user_locations (
+  id INT,
+  latitude DECIMAL(10,7),
+  longitude DECIMAL(10,7),
+  geohash VARCHAR
+) WITH (
+    KAFKA_TOPIC='user-locations', 
+    VALUE_FORMAT='JSON',
+    PARTITIONS=6
+);
+
+-- Stream of alerts when a user's geohash based location roughly 
+-- intersects a collection of merchants locations
+CREATE STREAM alerts_raw
+WITH (
+  KAFKA_TOPIC='alerts-raw', 
```

Comment:
This topic `alerts-raw` is empty (`user-locations` is not)

Reply:
@ybyzek on my test I see this:
```
PRINT 'alerts-raw' FROM BEGINNING;
Key format: ¯\_(ツ)_/¯ - no data processed
Value format: JSON or KAFKA_STRING
rowtime: 2022/01/26 14:40:24.691 Z, key: <null>, value: {"USER_ID":2,"USER_LATITUDE":14.5517401,"USER_LONGITUDE":121.0518652,"USER_GEOHASH":"wdw4f9","MERCHANT_ID":3}, partition: 4
rowtime: 2022/01/26 14:40:24.806 Z, key: <null>, value: {"USER_ID":2,"USER_LATITUDE":14.5500341,"USER_LONGITUDE":121.0555802,"USER_GEOHASH":"wdw4f8","MERCHANT_ID":2}, partition: 4
rowtime: 2022/01/26 14:40:24.806 Z, key: <null>, value: {"USER_ID":2,"USER_LATITUDE":14.5500341,"USER_LONGITUDE":121.0555802,"USER_GEOHASH":"wdw4f8","MERCHANT_ID":1}, partition: 4
rowtime: 2022/01/26 14:40:24.494 Z, key: <null>, value: {"USER_ID":1,"USER_LATITUDE":14.5472791,"USER_LONGITUDE":121.0475401,"USER_GEOHASH":"wdw4f8","MERCHANT_ID":2}, partition: 3
rowtime: 2022/01/26 14:40:24.494 Z, key: <null>, value: {"USER_ID":1,"USER_LATITUDE":14.5472791,"USER_LONGITUDE":121.0475401,"USER_GEOHASH":"wdw4f8","MERCHANT_ID":1}, partition: 3
rowtime: 2022/01/26 14:40:24.602 Z, key: <null>, value: {"USER_ID":2,"USER_LATITUDE":14.5486952,"USER_LONGITUDE":121.0521851,"USER_GEOHASH":"wdw4f8","MERCHANT_ID":2}, partition: 3
rowtime: 2022/01/26 14:40:24.602 Z, key: <null>, value: {"USER_ID":2,"USER_LATITUDE":14.5486952,"USER_LONGITUDE":121.0521851,"USER_GEOHASH":"wdw4f8","MERCHANT_ID":1}, partition: 3
```

Reply:
Now resolved